src/dreyfus_util.erl

```diff
@@ -166,6 +174,88 @@ time(Metric, {M, F, A}) when is_list(Metric) ->
         couch_stats:update_histogram([dreyfus | Metric],  Length)
     end.
 
+get_value_from_options(Key, Options) ->
+    case couch_util:get_value(Key, Options) of
+        undefined ->
+            Reason = binary_to_list(Key) ++ " must exist in Options.",
+            throw({bad_request, Reason});
+        Value -> Value
+    end.
+
+ensure_local_purge_docs(DbName, DDocs) ->
+    couch_util:with_db(DbName, fun(Db) ->
+        lists:foreach(fun(DDoc) ->
+            #doc{body = {Props}} = DDoc,
+            case couch_util:get_value(<<"indexes">>, Props) of
+                undefined -> false;
+                _ ->
+                    try dreyfus_index:design_doc_to_indexes(DDoc) of
+                        SIndexes -> ensure_local_purge_doc(Db, SIndexes)
+                    catch _:_ ->
+                        ok
+                    end
+            end
+        end, DDocs)
+    end).
+
+ensure_local_purge_doc(Db, SIndexes) ->
+    if SIndexes =/= [] ->
+        lists:map(fun(SIndex) ->
+            maybe_create_local_purge_doc(Db, SIndex)
+        end, SIndexes);
+    true -> ok end.
+
+maybe_create_local_purge_doc(Db, Index) ->
+    DocId = dreyfus_util:get_local_purge_doc_id(Index#index.sig),
+    case couch_db:open_doc(Db, DocId) of
+        {not_found, _} ->
+            {ok, DbPurgeSeq} = couch_db:get_purge_seq(Db),
+            DocContent = dreyfus_util:get_local_purge_doc_body(
+                Db, DocId, DbPurgeSeq, Index),
+            couch_db:update_doc(Db, DocContent, []);
+        _ ->
+            ok
+    end.
+
+maybe_create_local_purge_doc(Db, IndexPid, Index) ->
+    DocId = dreyfus_util:get_local_purge_doc_id(Index#index.sig),
+    case couch_db:open_doc(Db, DocId) of
+        {not_found, _} ->
+            {ok, DbPurgeSeq} = couch_db:get_purge_seq(Db),
+            clouseau_rpc:set_purge_seq(IndexPid, DbPurgeSeq),
+            DocContent = dreyfus_util:get_local_purge_doc_body(
+                Db, DocId, DbPurgeSeq, Index),
+            couch_db:update_doc(Db, DocContent, []);
+        _ ->
+            ok
+    end.
+
+get_local_purge_doc_id(Sig) ->
+    Version = "v" ++ config:get("purge", "version", "1") ++ "-",
+    ?l2b(?LOCAL_DOC_PREFIX ++ "purge-" ++ Version ++ "dreyfus-" ++ Sig).
+
+get_local_purge_doc_body(Db, LocalDocId, PurgeSeq, Index) ->
```

Comment:
Seems like we should probably move the verify_index_exists function to this module so that we don't have to look back and forth between the two to make sure names are the same in both places.